corehq/apps/registry/models.py

```diff
@@ -329,10 +329,17 @@ def schema_changed(self, user, new, old):
         )
 
     def data_accessed(self, user, domain, related_object, filters=None):
-        if not related_object or not hasattr(related_object, "doc_type"):
+        is_repeater = True if hasattr(related_object, 'repeater_type') else False
+        if (
+            not (related_object and hasattr(related_object, "doc_type"))
+            and not is_repeater
+        ):
             raise ValueError("Unexpected related object")
 
-        doc_type = getattr(related_object, 'base_doc', related_object.doc_type)
+        doc_type = (
+            getattr(related_object, 'base_doc', related_object.doc_type)
+            if not is_repeater else 'Repeater'
+        )
```

Comment:
I think you'll also need to do something about `related_object.get_id` below since I don't think SQL repeaters have that property. Possibly the best approach is to test if it's a SQL model and use `related_object.pk`

Reply:
A test in `registry/tests/test_logging.py` wouldn't go amiss.

Reply:
I have added it here https://github.com/dimagi/commcare-hq/pull/32102/commits/2752708a8303d4e3f93aee10dd009b09909d9757

Reply:
> A test in registry/tests/test_logging.py wouldn't go amiss.

I will add the test for the change. But I think it is also tested by other tests. I remember this test failing when passing SQL repeater with no `get_id` in it.

Reply:
@snopoke based on your suggestion, I used `related_object.pk` (https://github.com/dimagi/commcare-hq/pull/32102/commits/f5e52a39e7a6ba38240844e251881cdb718b944b) and have added tests for the change (https://github.com/dimagi/commcare-hq/pull/32102/commits/9e7ab5022975708f5e2032baebcfba6f37c32446)